# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBCursor.h
# Commit: 94ca5d86b3b0
# Full Hash: 94ca5d86b3b0c398294dd82d44194a5173c5aec6
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-01-09 21:39:42
# Regressor Bug: 1600906
# File Overlap Count: 1
# Description:
#   Bug 1600906 - Use scoped enums in IDBCursor. r=dom-workers-and-storage-reviewers,ytausky
#   
#   Depends on D57990
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D57991
# ==============================================================================

diff -r 801f97f67366 -r 94ca5d86b3b0 dom/indexedDB/IDBCursor.h
--- a/dom/indexedDB/IDBCursor.h	Thu Jan 09 14:58:17 2020 +0000
+++ b/dom/indexedDB/IDBCursor.h	Mon Jan 06 14:19:39 2020 +0000
@@ -38,24 +38,24 @@
 // distinctions.
 class IDBCursor final : public nsISupports, public nsWrapperCache {
  public:
-  typedef indexedDB::Key Key;
-  typedef indexedDB::StructuredCloneReadInfo StructuredCloneReadInfo;
+  using Key = indexedDB::Key;
+  using StructuredCloneReadInfo = indexedDB::StructuredCloneReadInfo;
 
-  enum Direction {
-    NEXT = 0,
-    NEXT_UNIQUE,
-    PREV,
-    PREV_UNIQUE,
+  enum struct Direction {
+    Next = 0,
+    NextUnique,
+    Prev,
+    PrevUnique,
 
     // Only needed for IPC serialization helper, should never be used in code.
-    DIRECTION_INVALID
+    Invalid
   };
 
-  enum Type {
-    Type_ObjectStore,
-    Type_ObjectStoreKey,
-    Type_Index,
-    Type_IndexKey,
+  enum struct Type {
+    ObjectStore,
+    ObjectStoreKey,
+    Index,
+    IndexKey,
   };
 
  private: